### PR TITLE
Improve watch API

### PR DESCRIPTION
**Pull request by kostko**

### CODE REVIEW

File: src/core/components/base.ts [hunk 1]
```diff
@@ -379,10 +411,18 @@ export abstract class ComponentBase {
      * @param context Function which returns the context to watch
      * @param content Function to run on changes
      */
-    public watchCollection(context: WatchExpression, content: ComputationFunction): void {
```
Comment (lukaw3d) [inline src/core/components/base.ts]: Oh; watchCollection wasn't un-watchable before

File: src/core/components/base.ts [hunk 2]
```diff
@@ -355,21 +376,32 @@ export abstract class ComponentBase {
      * @param objectEquality Should `angular.equals` be used for comparisons
      * @returns A function that unregisters the bound expression
```
Comment (hadalin) [inline src/core/components/base.ts]: This doc seems out of sync. `Computation` is now returned.

**Comment by hadalin** [comment]
Typo https://github.com/kostko/resolwe-js/blob/53d26575cc1f14f1084b1849bb037915a464a3b8/src/core/components/base.ts#L228

File: src/core/components/base.ts [hunk 3]
```diff
@@ -379,10 +411,18 @@ export abstract class ComponentBase {
      * @param context Function which returns the context to watch
      * @param content Function to run on changes
```
Comment (hadalin) [inline src/core/components/base.ts]: Doc out of sync. Returns `Computation`.

File: src/core/components/base.ts (hunk 2 above)
Comment (kostko) [inline src/core/components/base.ts]: Thanks for pointing this out, should be fixed in #30.

**Pull request merged**